Automattic/jetpack
# Packages: Fix connection tests and allow them to run in CI

Currently, connection package tests rely on WP_Mock, which requires PHP >= 7.1. That doesn't work for us, since we have to support 5.6 for development.

This PR updates the tests so they use PHPmock (like the logo package) and updates the package to use the same infrastructure so tests run on CI.

#### Changes proposed in this Pull Request:
* Use php-mock instead of wp_mock
* Migrate tests from wp_mock to phpmock
* Add phpunit script for CI
* Changing the file structure and the tests bootstrap to allow for CI to run the tests.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Part of the Jetpack DNA project - p1HpG7-70O-p2

#### Testing instructions:
* Checkout this branch.
* Run `rm -rf ./vendor` inside Jetpack.
* Run `composer install` inside Jetpack.
* Navigate to `packages/connection`
* Run `composer phpunit`.
* Verify that runs `composer install` internally, and then runs the package tests.
* Verify the package tests pass.
* Run `composer phpunit` again.
* Verify the second time is fast because the dependencies are already installed.
* Verify CI passes on this PR and 100% is green.

#### Proposed changelog entry for your changes:
* Migrate the connection package tests to support older PHP versions
